apps/prairielearn/src/ee/lib/aiQuestionGeneration.ts

```diff
@@ -362,39 +441,80 @@ Keep in mind you are not just generating an example; you are generating an actua
     user: openAiUserFromAuthn(authnUserId),
   });
 
-  extractFromCompletion(completion, job);
-  job.data['generation'] = completion.choices[0].message.content;
-  job.data['completion'] = completion;
+  const results = extractFromCompletion(completion, job);
 
-  const html = job?.data?.html || originalHTML;
+  const html = results?.html || originalHTML;
 
-  if (saveInitialErrors) {
-    job.data['initialGenerationErrors'] = [];
-  }
-  job.data['finalGenerationErrors'] = [];
+  let errors: string[] = [];
 
   if (html && typeof html === 'string') {
-    const errors = validateHTML(html, false, !!job?.data?.python);
-    if (saveInitialErrors) {
-      job.data['initialGenerationErrors'] = errors;
+    errors = validateHTML(html, false, !!results?.python);
+  }
+
+  if (userId !== undefined && hasCoursePermissionEdit !== undefined) {
```

Comment:
When will these ever be undefined? The types don't allow that.

Reply:
removed check in 8f91ed5 as it no longer applies